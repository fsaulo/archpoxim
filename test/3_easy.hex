0xDC00001F
0x00000000
0x80000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00000000
0x00201234
0x4C410004
0x10620804
0x10C52101
0x10E53400
0x87E10001
0x54E50000
0x58E50000
0x29484FC7
0x2FE83A89
0x494A0001
0x7C000000
0x6C200090
0x68400024
0x70200048
0x00000000
0x68400024
0x70200048
0x60400090
0x74200024
0x64400049
0x5C020000
0xD0000000
0x7C000000
0x03C07FFC
0x78000008
0xE7FFFFF1
0xFC000000
0x00000000
