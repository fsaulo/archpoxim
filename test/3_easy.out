[START OF SIMULATION]
0x00000000:	bun 31                   	PC=0x00000080
0x00000080:	mov sp,32764             	SP=0x00007FFC
0x00000084:	call [r0+8]              	PC=0x00000020,MEM[0x00007FFC]=0x00000088
0x00000020:	mov r1,4660              	R1=0x00001234
0x00000024:	subi r2,r1,4             	R2=R1-0x00000004=0x00001230,SR=0x00000000
0x00000028:	mul r4,r3,r2,r1          	R4:R3=R2*R1=0x00000000014B11C0,SR=0x00000000
0x0000002C:	sll r6,r5,r4,1           	R6:R5=R6:R4<<2=0x0000000000000000,SR=0x00000040
0x00000030:	div r0,r7,r5,r6          	R0=R5%R6=0x00000000,R7=R5/R6=0x00000000,SR=0x00000060
0x00000034:	sbr sr[1]                	SR=0x00000062
0x00000038:	divi r7,r5,0             	R7=R5/0x00000000=0x00000000,SR=0x00000062
[SOFTWARE INTERRUPTION]
0x00000008:	reti                     	IPC=MEM[0x00007FF0]=0x00000000,CR=MEM[0x00007FF4]=0x00000000,PC=MEM[0x00007FF8]=0x0000003C
0x0000003C:	modi r7,r5,0             	R7=R5%0x00000000=0x00000000,SR=0x00000062
[SOFTWARE INTERRUPTION]
0x00000008:	reti                     	IPC=MEM[0x00007FF0]=0x00000000,CR=MEM[0x00007FF4]=0x00000000,PC=MEM[0x00007FF8]=0x00000040
0x00000040:	push sr,r7,r8,r9,r10     	MEM[0x00007FF8]{0x00000062,0x00000000,0x00000000,0x00000000,0x00000000}={SR,R7,R8,R9,R10}
0x00000044:	pop r10,r9,r8,r7,sr      	{R10,R9,R8,R7,SR}=MEM[0x00007FE4]{0x00000000,0x00000000,0x00000000,0x00000000,0x00000062}
0x00000048:	addi r10,r10,1           	R10=R10+0x00000001=0x00000001,SR=0x00000022
0x0000004C:	ret                      	PC=MEM[0x00007FFC]=0x00000088
0x00000088:	call -15                 	PC=0x00000050,MEM[0x00007FFC]=0x0000008C
0x00000050:	s8 [r0+144],r1           	MEM[0x00000090]=R1=0x34
0x00000054:	l32 r2,[r0+36]           	R2=MEM[0x00000090]=0x34000000
0x00000058:	s16 [r0+72],r1           	MEM[0x00000090]=R1=0x1234
0x0000005C:	mov r0,0                 	R0=0x00000000
0x00000060:	l32 r2,[r0+36]           	R2=MEM[0x00000090]=0x12340000
0x00000064:	s16 [r0+72],r1           	MEM[0x00000090]=R1=0x1234
0x00000068:	l8 r2,[r0+144]           	R2=MEM[0x00000090]=0x12
0x0000006C:	s32 [r0+36],r1           	MEM[0x00000090]=R1=0x00001234
0x00000070:	l16 r2,[r0+73]           	R2=MEM[0x00000092]=0x1234
0x00000074:	cmpi r2,0                	SR=0x00000022
0x00000078:	bne 0                    	PC=0x0000007C
0x0000007C:	ret                      	PC=MEM[0x00007FFC]=0x0000008C
0x0000008C:	int 0                    	CR=0x00000000,PC=0x00000000
[END OF SIMULATION]
